FirebaseAppDistributionInternal/Tests/Unit/AppDistributionApiServiceTests.swift

```diff
@@ -13,6 +13,156 @@
+    options.apiKey = "A7a8Ff2UsWT3r5lOg22fFFkVwZClxc2MsvfPPFS"
+    FirebaseApp.configure(name: "app-distribution-test-app", options: options)
+    _ = FirebaseApp.app(name: "app-distribution-test-app")
+  }
+
+  // MARK: - Test generateAuthToken
+
+  func testGenerateAuthTokenSuccess() {
+    let installations = FakeInstallations(testCase: .success)
+
+    let expectation = XCTestExpectation(description: "Generate auth token succeeds")
+
+    AppDistributionApiService.generateAuthToken(
+      installations: installations,
+      completion: { identifier, authTokenResult, error in
+        XCTAssertNotNil(identifier)
+        XCTAssertNotNil(authTokenResult)
+        XCTAssertNil(error)
+        expectation.fulfill()
+      }
+    )
+
+    wait(for: [expectation], timeout: 5)
+  }
+
+  func testGenerateAuthTokenAuthTokenFailure() {
+    let installations = FakeInstallations(testCase: .authTokenFailure)
+    let expectation =
+      XCTestExpectation(description: "Generate auth token fails to generate auth token.")
+
+    AppDistributionApiService.generateAuthToken(
+      installations: installations,
+      completion: { identifier, authTokenResult, error in
+        let nserror = error as? NSError
+        XCTAssertNil(identifier)
+        XCTAssertNil(authTokenResult)
+        XCTAssertNotNil(error)
+        XCTAssertEqual(nserror?.code, AppDistributionApiError.ApiTokenGenerationFailure.rawValue)
+        expectation.fulfill()
+      }
+    )
+
+    wait(for: [expectation], timeout: 5)
+  }
+
+  func testGenerateAuthTokenInstallationIDFailure() {
+    let installations = FakeInstallations(testCase: .installationIDFailure)
+    let expectation = XCTestExpectation(description: "Generate auth token fails to find ID.")
+
+    AppDistributionApiService.generateAuthToken(
+      installations: installations,
+      completion: { identifier, authTokenResult, error in
+        let nserror = error as? NSError
+        XCTAssertNil(identifier)
+        XCTAssertNil(authTokenResult)
+        XCTAssertNotNil(error)
+        XCTAssertEqual(
+          nserror?.code,
+          AppDistributionApiError.ApiInstallationIdentifierError.rawValue
+        )
+        expectation.fulfill()
+      }
+    )
+
+    wait(for: [expectation], timeout: 5)
+  }
+
+  // MARK: - Test fetchReleases
+
+  func testFetchReleasesSuccess() {
+    let app = FirebaseApp.app(name: "app-distribution-test-app")!
+    let installations = FakeInstallations(testCase: .success)
+    let urlSession = URLSessionMock(testCase: .success)
+
+    let expectation = XCTestExpectation(description: "Fetch releases succeeds with two releases.")
+
+    AppDistributionApiService.fetchReleases(
+      app: app,
+      installations: installations,
+      urlSession: urlSession,
+      completion: { releases, error in
+        XCTAssertNotNil(releases)
+        XCTAssertNil(error)
+        XCTAssertEqual(releases?.count, 2)
```

Comment:
Should we also assert the value of the releases variable?

Reply:
Added